### ISSUE TITLE
Tiger Tree Hash - API / Code Mismatch

### ISSUE THREAD

**Issue by andreas-kupries** [rhash/RHash#143]
Header and code disagree on the size of the resulting hash/digest.

```
tth.h:23: void rhash_tth_final(tth_ctx* ctx, unsigned char result[64]);
tth.c:93: void rhash_tth_final(tth_ctx* ctx, unsigned char result[24])
..................................................................^
```
The header claims a digest size of `64` bytes, the implementation uses `24` only.
Given that TTH is a hash tree based on Tiger, and Tiger's hash size is `24`, I suspect that the typo is in the header/declaration.
Maybe define a `tth_hash_length` (or `tth_has_size`) macro ?

**Comment by rhash** [comment]
Thanks for reporting this issue! It's fixed by 821b030830771dcce4b660cd1cfc463d8374f2fa.

**Comment by andreas-kupries** [comment]
You are welcome. (Do I (have to) close the ticket now ? What with the `done` label and still open).

FYI, the issue was originally found when testing https://github.com/andreas-kupries/ankh and the hash results were unstable in the lower bytes (i.e. random memory data).
